src/System.IO.Compression.Brotli/System/IO/Compression/Brotli/BrotliPrimitives.cs

```diff
@@ -16,41 +17,64 @@ public struct BrotliPrimitives
         const int defQuality = 11;
         const int defLgWin = 24;
 
-        public static bool Compress(ReadOnlySpan<byte> source, Span<byte> destination, out nuint bytesConsumed, out nuint bytesWritten)
+        public static int GetMaximumCompressedSize(int input_size)
         {
-            return Compress(source, destination, out bytesConsumed, out bytesWritten, defQuality, defLgWin);
+            if (input_size == 0) return 1;
+            int num_large_blocks = input_size >> 24;
+            int tail = input_size & 0xFFFFFF;
+            int tail_overhead = (tail > (1 << 20)) ? 4 : 3;
+            int overhead = 2 + (4 * num_large_blocks) + tail_overhead + 1;
+            int result = input_size + overhead;
+            return (result < input_size) ? input_size : result;
         }
 
-        public static bool Compress(ReadOnlySpan<byte> source, Span<byte> destination, out nuint bytesConsumed, out nuint bytesWritten, int quality, int lgwin)
+        public static bool Compress(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesConsumed, out int bytesWritten, int quality=defQuality, int lgwin=defLgWin, BrotliEncoderMode encMode=BrotliEncoderMode.Generic)
```

Comment:
We have just discussed(https://github.com/dotnet/corefxlab/pull/1598#pullrequestreview-42784850). I can do a simple signature, but it'll gives less opportunities to configure compress and in some cases it can be less effective. Should I delete added parameters? Thanks